src/main/java/net/sf/jabref/gui/keyboard/KeyBindingsDialogViewModel.java

```diff
@@ -0,0 +1,210 @@
+    }
+
+    /**
+     * Read all keybindings from the keybinding repository and create table keybinding
+     * models for them
+     */
+    private void populateTable() {
+        TreeItem<KeyBindingWrapper> root = new TreeItem<>(new KeyBindingWrapper(null, "", KeyBindingCategory.FILE));
+        for (KeyBindingCategory category : KeyBindingCategory.values()) {
+            TreeItem<KeyBindingWrapper> categoryItem = new TreeItem<>(new KeyBindingWrapper(null, "", category));
+            keyBindingRepository.getKeyBindings().forEach((keyBinding, bind) -> {
+                if (keyBinding.getCategory() == category) {
+                    KeyBindingWrapper keyBindWrapper = new KeyBindingWrapper(keyBinding, bind,
+                            keyBinding.getCategory());
+                    TreeItem<KeyBindingWrapper> keyBindTreeItem = new TreeItem<>(keyBindWrapper);
+                    categoryItem.getChildren().add(keyBindTreeItem);
+                }
+            });
+            categoryItem.setExpanded(true);
+            root.getChildren().add(categoryItem);
+        }
+        root.setExpanded(true);
+        keyBindingsTable.setRoot(root);
+    }
+
+    private void bindColumnValues() {
+        actionColumn
+                .setCellValueFactory(cellData -> new SimpleStringProperty(cellData.getValue().getValue().getName()));
+        shortcutColumn.setCellValueFactory(cellData -> cellData.getValue().getValue().bindProperty());
+    }
+
+    private void registerKeyEvents() {
+        keyBindingsTable.setOnKeyPressed(evt -> grabKey(evt));
+    }
+
+    private Stage getDialogStage() {
+        return (Stage) closeButton.getScene().getWindow();
+    }
+
+    @FXML
+    private void closeDialog() {
+        Stage stage = getDialogStage();
+        stage.close();
+    }
+
+    @FXML
+    private void saveKeyBindingsAndCloseDialog() {
+        Globals.getKeyPrefs().setNewKeyBindings(keyBindingRepository.getKeyBindings());
+
+        // show success message
+        FXAlert alert = new FXAlert(AlertType.INFORMATION, Localization.lang("Key bindings changed"));
+        alert.setHeaderText(null);
+        alert.setContentText(Localization.lang("Your new key bindings have been stored.") + '\n'
+                + Localization.lang("You must restart JabRef for the new key bindings to work properly."));
+        alert.getButtonTypes().clear();
+        alert.getButtonTypes().add(ButtonType.CLOSE);
+        alert.showAndWait().ifPresent(response -> {
+            if (response == ButtonType.CLOSE) {
+                closeDialog();
+            }
+        });
+    }
+
+    @FXML
+    private void setDefaultBindings() {
+        ObservableList<TreeItem<KeyBindingWrapper>> selectedKeyBindingWrapperItems = keyBindingsTable
+                .getSelectionModel().getSelectedItems();
+        if (selectedKeyBindingWrapperItems.isEmpty()) {
+            FXAlert alert = new FXAlert(AlertType.INFORMATION, Localization.lang("Resetting all key bindings"));
```

Comment:
Similar to above, move to `boolean Dialogs.showConfirmation( title, text, buttonText)`